sql/catalyst/src/main/scala/org/apache/spark/sql/internal/StaticSQLConf.scala

```diff
@@ -145,9 +145,17 @@ object StaticSQLConf {
         "cause longer waiting for other broadcasting. Also, increasing parallelism may " +
         "cause memory problem.")
       .intConf
-      .checkValue(thres => thres > 0 && thres <= 128, "The threshold must be in [0,128].")
+      .checkValue(thres => thres > 0 && thres <= 128, "The threshold must be in (0,128].")
       .createWithDefault(128)
 
+  val SUBQUERY_MAX_THREAD_THRESHOLD =
```

Comment:
This is actually a different change right?

Reply:
It should also be a static configuration since we can only change it at startup.

Reply:
1. This is just opening as a configration to make the change testable

2. This is part of `StaticSQLConf` which is defined at startup, is there any other mechanism to define static conf.?

Reply:
If its a static conf, then why isn't your unite test failing? Moreover, if its static then setting it in your test probably does not have any effect because we use the same JVM/SparkContext to run most tests, the chances are pretty high that it has been set before.

Reply:
i see. as per documentation in https://github.com/apache/spark/blob/705fc6ad9328c1092299b83071b6ec3b1d6f9c4d/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/StaticSQLConf.scala#L26 its should not modified. I followed the same way `BroadcastExchangeExec` creates executionContext. My initial guess is executionContext is not created till first subquery hence it work for UT. i will further investigate and get back with analysis.